### PR TITLE
PaywallView support

### PR DESCRIPTION
**Pull request by vegaro**
Added PaywallView for iOS and Android. Note this requires bumping the compile version in android to 34 if you're using purchases_ui_flutter.

### CODE REVIEW

File: purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt
Comment (vegaro) [inline purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt:28]: this requires a phc release so doing it this way for now for the sake of shipping

File: purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt
Comment (tonidero) [inline purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt:8]: Hmm I wonder if we should rename this class to `RNPaywallView` instead... but I'm ok with this for now.

**Review by tonidero — APPROVED** [review]
Looks good! I would hold until we have the iOS part ready, or merge into a different base branch though, so it's not broken in iOS.

File: android/build.gradle
Comment (tonidero) [inline android/build.gradle:29]: I'm wondering if this was needed or if it was only on the `purchases_ui_flutter` library... However, I think it shouldn't hurt.

File: android/build.gradle
Comment (tonidero) [inline android/build.gradle:13]: While this shouldn't cause any issues, was this needed? So we can extract to a different PR (In purchases Android we currently have 8.1.1)

File: purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallViewFactory.kt
Comment (tonidero) [inline purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallViewFactory.kt:8]: We could make this internal I think

File: purchases_ui_flutter/lib/views/paywall_view.dart
Comment (tonidero) [inline purchases_ui_flutter/lib/views/paywall_view.dart:28]: These should be private (I made this change in #993)

File: purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt
Comment (tonidero) [inline purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt:6]: Total nitpick, but these imports need sorting :P

File: purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift
Comment (tonidero) [inline purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift:38]: We are not getting and setting the offering id in iOS

**Review by tonidero — COMMENTED** [review]
Left some comments... Ideally we would try to avoid upgrading as many things as possible in this PR, specially, I think we shouldn't touch the main library and the gradle plugin if possible. We also need to fix using a specific offering in iOS. Other than that look good

File: android/build.gradle
Comment (tonidero) [reply android/build.gradle]: I reverted this in my followup commits

File: android/build.gradle
Comment (tonidero) [reply android/build.gradle]: Same, I reverted this in my followup commits

File: purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallViewFactory.kt
Comment (tonidero) [reply purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallViewFactory.kt]: Done in my followup commits

File: purchases_ui_flutter/lib/views/paywall_view.dart
Comment (tonidero) [reply purchases_ui_flutter/lib/views/paywall_view.dart]: Done in my followup commits

File: purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift
Comment (tonidero) [reply purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift:38]: Done in https://github.com/RevenueCat/purchases-flutter/pull/954/commits/7c0c6efce95382e5ba326c592fb569c4aa0d024a

File: revenuecat_examples/purchase_tester/lib/src/paywall.dart
Comment (vegaro) [inline revenuecat_examples/purchase_tester/lib/src/paywall.dart:24]: let's remove that comment `// Wrap your body content with SafeArea`

File: revenuecat_examples/purchase_tester/lib/src/upsell.dart
Comment (vegaro) [inline revenuecat_examples/purchase_tester/lib/src/upsell.dart:116]: I think this needs some formatting

**Review by vegaro — COMMENTED** [review]
I can't approve it because it's my PR but looks good

File: purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift
Comment (tonidero) [reply purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift:38]: Yup

**Pull request merged**
